src/integration/pyboy_integration.py

```diff
@@ -28,4 +30,41 @@ def pyboy_tick(button=""):
     pyboy.screen.image.resize((480, 432), 0).save("data/image.png")
     with open("data/state_file.state", "wb") as f:
         pyboy.save_state(f)
+
+    game_info = GameInformation.from_pyboy(pyboy)
+
     pyboy.stop()
+
+    return game_info
+
+
+@dataclass
+class GameInformation:
+    player_name: str
+
+    @classmethod
+    def from_pyboy(cls, pyboy) -> Self:
+        return GameInformation(
+            player_name=_bytes_as_gen1_string(pyboy.memory[0xD158:0xD158 + 0xB]),
+        )
+
+    def as_markdown(self) -> str:
+        return f"""
+# Game Data
+
+Name: {self.player_name}
+"""
+
+
+STRING_TERMINATOR = 80
+ASCII_DELTA = 63
+
+
+def _bytes_as_gen1_string(data) -> str:
```

Comment:
Reckon we could do some sick list comprehension here

```
def _bytes_as_gen1_string(data) -> str:
    return ''.join(chr(byte - ASCII_DELTA) for byte in data if byte != STRING_TERMINATOR)
```

Reply:
yeah this was ripped directly from the PR you found